File: 945-minimum-increment-to-make-array-unique/main.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        
        nums.sort()
    
        increments = 0
        
        for i in range(1, len(nums)):
            if nums[i] <= nums[i - 1]:
                needed = nums[i - 1] + 1 - nums[i]
                nums[i] += needed
                increments += needed
        
        return increments

File: 945-minimum-increment-to-make-array-unique/main.rs
impl Solution {
    pub fn min_increment_for_unique(nums: Vec<i32>) -> i32 {
        
        let mut sorted_nums = nums.clone();
        sorted_nums.sort();

        let mut increments = 0;
        for i in 1..sorted_nums.len() {
            if sorted_nums[i] <= sorted_nums[i - 1] {
                let needed = sorted_nums[i - 1] + 1 - sorted_nums[i];
                sorted_nums[i] += needed;
                increments += needed;
            }
        }

        increments
    }
}
